OptimizelySDK.Tests/UtilsTests/EventTagUtilsTest.cs

```diff
@@ -0,0 +1,208 @@
+    class EventTagUtilsTest
+    {
+        private string TestUserId = string.Empty;
+        private ProjectConfig Config;
+        private EventBuilder EventBuilder;
+
+        [TestFixtureSetUp]
+        public void Setup()
+        {
+            TestUserId = "testUserId";
+            var logger = new NoOpLogger();
+            Config = ProjectConfig.Create(TestData.Datafile, logger, new ErrorHandler.NoOpErrorHandler());
+            EventBuilder = new EventBuilder(new Bucketer(logger));
+        }
+
+        [Test]
+        public void TestGetRevenueValue()
+        {
+            var expectedValue = 42;
+            var expectedValue2 = 100;
+            var validTag = new Dictionary<string, object>() {
+                { "revenue", 42 }
+            };
+            var validTag2 = new Dictionary<string, object>() {
+                { "revenue", 100 }
+            };
+
+            var invalidTag = new Dictionary<string, object>() {
+                { "abc", 42 }
+            };
+            var nullValue = new Dictionary<string, object>() {
+                { "revenue", null }
+            };
+            var invalidValue = new Dictionary<string, object>() {
+                { "revenue", 42.5 }
+            };
+
+            // Invalid data.
+            Assert.Null(EventTagUtils.GetRevenueValue(null));
+            Assert.Null(EventTagUtils.GetRevenueValue(invalidTag));
+            Assert.Null(EventTagUtils.GetRevenueValue(nullValue));
+            Assert.Null(EventTagUtils.GetRevenueValue(invalidValue));
+
+            // Valid data.
+            Assert.AreEqual(EventTagUtils.GetRevenueValue(validTag), expectedValue);
+            Assert.AreEqual(EventTagUtils.GetRevenueValue(validTag2), expectedValue2);
+        }
+
+        [Test]
+        public void TestGetEventValue()
+        {
+            int expectedValue = 42;
+            float expectedValue2 = 42.5F;
+            double expectedValue3 = 42.52;
+
+            var validTag = new Dictionary<string, object>() {
+                { "value", 42 }
+            };
+
+            var validTag2 = new Dictionary<string, object>() {
+                { "value", 42.5 }
+            };
+
+            var validTag3 = new Dictionary<string, object>() {
+                { "value", 42.52 }
+            };
+
+            var invalidTag = new Dictionary<string, object>() {
+                { "abc", 42 }
+            };
+            var nullValue = new Dictionary<string, object>() {
+                { "value", null }
+            };
+            var validTagStr = new Dictionary<string, object>() {
+                { "value", "42" }
+            };
+            var validTagStr1 = new Dictionary<string, object>() {
+                { "value", "42.3" }
+            };
+
+            // Invalid data.
+            Assert.Null(EventTagUtils.GetNumericValue(null));
+            Assert.Null(EventTagUtils.GetNumericValue(invalidTag));
+            Assert.Null(EventTagUtils.GetNumericValue(nullValue));
```

Comment:
Also include the following invalid cases:

{ "value" : "1,234" }
{ "value" : int [] }
{ "value" : true }

These invalid cases should return `null`

Reply:
We are using tryparse, and for the first case "1,234", tryparse is able to parse 1,234 into 1234.0. If u say I can check it using regex? Please suggest.